cpp/src/slice2cpp/Gen.cpp

```diff
@@ -543,45 +543,70 @@ writeOpDocSummary(Output& out, const OperationPtr& op, const CommentPtr& doc, Op
 
 // Returns the client-side result type for an operation - can be void, a single type, or a tuple.
 string
-createResultType(const OperationPtr& p, const string& scope, int useWstring)
+createOutgoingAsyncTypeParam(const vector<string>& elements)
 {
-    assert(useWstring == 0 || useWstring == TypeContextUseWstring);
+    if (elements.size() == 0)
+    {
+        return "void";
+    }
+    else if (elements.size() == 1)
+    {
+        return elements.front();
+    }
+    else
+    {
+        ostringstream os;
+        Output out(os);
+        out << "::std::tuple" << sabrk;
+        for (const auto& element : elements)
+        {
+            out << element;
+        }
+        out << eabrk;
+        return os.str();
+    }
+}
 
+vector<string>
+createOutgoingAsyncParams(const OperationPtr& p, const string& scope, int typeContext)
```

Comment:
Would be nice to have a comment here. It sounds similar to the one above it, but takes a very different parameters.

Reply:
added comment.